### PR TITLE
feat: submit fee estimates in oracle

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

### CODE REVIEW

File: bitcoin/src/lib.rs [hunk 1]
```diff
@@ -150,12 +150,9 @@ impl LockedTransaction {
 
 fn get_exponential_backoff() -> ExponentialBackoff {
     ExponentialBackoff {
-        max_elapsed_time: Some(Duration::from_secs(24 * 60 * 60)),
-        max_interval: Duration::from_secs(5 * 60), // wait at most 5 minutes before retrying
-        initial_interval: Duration::from_secs(1),
-        current_interval: Duration::from_secs(1),
```
Comment (sander2) [inline bitcoin/src/lib.rs:156]: Why did you remove this? Imo the defaults are not suitable for us - it will retry a ton of times in the first few seconds. And also I prefer to have the configuration explicit.

File: oracle/src/main.rs [hunk 2]
```diff
@@ -79,55 +188,44 @@ async fn main() -> Result<(), Error> {
...
-        .set_exchange_rate(exchange_rate)
-        .await;
-
-        if let Err(e) = result {
-            error!("Error: {}", e.to_string());
+        let parachain_rpc =
+            InterBtcParachain::from_url_with_retry(&opts.btc_parachain_url, signer.clone(), opts.connection_timeout_ms)
+                .await?;
+
+        if let Err(err) = try_join!(
```
Comment (sander2) [inline oracle/src/main.rs:215]: Is there a reason for using `try_join` over `join`? When one call fails, don't we still want to continue executing the second call?

Comment (gregdhill) [reply oracle/src/main.rs:215]: It should return on the first `Err`, see: https://docs.rs/tokio/1.10.0/tokio/macro.try_join.html

Comment (gregdhill) [reply oracle/src/main.rs:215]: Ah wait, I misread your comment... Yeah we probably do want `join!`.

File: oracle/src/main.rs [hunk 3]
```diff
@@ -79,55 +197,46 @@ async fn main() -> Result<(), Error> {
...
-            Duration::from_millis(opts.connection_timeout_ms),
-        )
-        .await?
-        .set_exchange_rate(exchange_rate)
-        .await;
-
-        if let Err(e) = result {
-            error!("Error: {}", e.to_string());
+        if left.is_err() || right.is_err() {
+            return Err(Error::Shutdown);
```
Comment (sander2) [inline oracle/src/main.rs:237]: Why exit on failure? Imo the oracle should never quit - it should keep running forever, without depending on external restarts

Comment (gregdhill) [reply oracle/src/main.rs:237]: This will only return after the `backoff` period has expired, in which case something is seriously wrong (can't reach CoinGecko / Blockstream) and we should shut down.

Comment (gregdhill) [reply oracle/src/main.rs:231]: Just pushed a new commit, I actually saw no reason not to use `backoff` here instead and then error.

Comment (sander2) [reply oracle/src/main.rs:231]: I still think continuing to retry is better, but it doesn't have to block this pr..

**Review by sander2 — APPROVED** [review]

File: oracle/src/main.rs (hunk 3 above)
Comment (gregdhill) [reply oracle/src/main.rs:231]: Moved to https://github.com/interlay/interbtc-clients/issues/164
